File: app/src/main/java/com/yhjoo/dochef/adapter/CommentListAdapter.java
package com.yhjoo.dochef.adapter;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.yhjoo.dochef.R;
import com.yhjoo.dochef.model.Comment;
import com.yhjoo.dochef.utils.Utils;

public class CommentListAdapter extends BaseQuickAdapter<Comment, BaseViewHolder> {
    String mUserID = "";

    public CommentListAdapter(String userID) {
        super(R.layout.li_comment);
        mUserID = userID;
    }

    @Override
    protected void convert(BaseViewHolder helper, Comment item) {
        helper.setText(R.id.li_comment_nickname, item.getNickName());
        helper.setText(R.id.li_comment_contents, item.getContents());
        helper.setText(R.id.li_comment_date, Utils.convertMillisToText(item.getDateTime()));
        helper.setVisible(R.id.li_comment_other, item.getUserID().equals(mUserID));
        helper.addOnClickListener(R.id.li_comment_other);
    }
}

File: app/src/main/java/com/yhjoo/dochef/adapter/CommentListAdapter.kt
package com.yhjoo.dochef.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder
import com.yhjoo.dochef.R
import com.yhjoo.dochef.model.Comment
import com.yhjoo.dochef.utilities.Utils

class CommentListAdapter(private var mUserID: String) :
    BaseQuickAdapter<Comment, BaseViewHolder>(R.layout.comment_item) {
    override fun convert(helper: BaseViewHolder, item: Comment) {
        helper.setText(R.id.comment_nickname, item.nickName)
        helper.setText(R.id.comment_contents, item.contents)
        helper.setText(R.id.comment_date, Utils.convertMillisToText(item.dateTime))
        helper.setVisible(R.id.comment_other, item.userID == mUserID)
        helper.addOnClickListener(R.id.comment_other)
    }
}